packages/application-shell/src/components/navbar/navbar.js

```diff
@@ -534,6 +548,7 @@ export class DataMenu extends React.PureComponent {
                           useFullRedirectsForLinks={
                             this.props.useFullRedirectsForLinks
                           }
+                          trackingEvent={submenu.trackingEvent}
```

Comment:
This has several implications, so I wouldn't define this in the menu config.

Instead, how about passing a function prop from the `ApplicationShell` component?
Something like `onMenuItemClick` or something, then you are free to handle the event the way you want.
The callback function can get the `gtm` object as an argument, together with the `event`.

Reply:
Took the approach you suggested. Can you, please, take a look? Did I get your idea correctly, or you meant something different?